lib/iris/_data_manager.py

```diff
@@ -242,7 +267,8 @@ def data(self, data):
         else:
             if not ma.isMaskedArray(data):
                 # Coerce input data to ndarray (including ndarray subclasses).
-                data = np.asarray(data)
+                if not dataless:
+                    data = np.asarray(data)
```

Comment:
@ESadek-MO We shouldn't need this defensive code for the dataless case i.e., we shouldn't get here.